### PR TITLE
Channel based bans

### PR DESCRIPTION
**Pull request by S-S-X**
Implements #29 

Allows banned players to read channel but disallows sending messages.

Might work, didn't test, didn't try if minetest can load this. So _might_ work but I've no idea really...

Requires enabling extension with following configuration (minetest conf file):
```lua
beerchat.enable_ban = true
```

Commands:
| Command                               | What it does
--------------------------------- | -----------------
| `/channel_ban player_name [player_name ...]` | Add bans for players on current channel
| `/channel_unban [player_name ...]` | Removes bans for players on current channel
| `/channel_unban` | Lists banned names on current channel

Missing stuff:
* Should allow channel name with `/channel_ban #main Sam`, currently always operates on active channel.
* Same for `/channel_unban #main Sam`
* Maybe add optional flag to also ban reading channel instead of just disallowing writing to channel.
   Reading is allowed simply to allow reading public info channels and currently this is most important actual use case for channel bans.
   In other words to prevent writing to some public channels like main chat channel without actually removing player from that channel.
   
Does not check if player exists, this should allow also banning external names. Not sure if this is really needed because there's also `/remote_mute` from another extension (bit different but still similar).

### CODE REVIEW

**Comment by S-S-X** [comment]
Might work.

Missing stuff:
* Should allow channel name with `/channel_ban #main Sam`, currently always operates on active channel.
* Same for `/channel_unban #main Sam`
* Maybe add optional flag to also ban reading channel instead of just disallowing writing to channel.
   Reading is allowed simply to allow reading public info channels and currently this is most important actual use case for channel bans.
   In other words to prevent writing to some public channels like main chat channel without actually removing player from that channel.

**Comment by S-S-X** [comment]
Plan was to extend this later with option to also ban reading channel / joining channel. Possibly also include `/channel_kick` here.

But not here, not going to add any of those additional features until fate of this PR is decided. This is what is needed to solve real issues, imaginary issues can be solved later.

**Review by wsor4035 — APPROVED** [review]
